src/main/java/com/es/phoneshop/dao/impl/GenericDao.java

```diff
@@ -0,0 +1,53 @@
+package com.es.phoneshop.dao.impl;
+
+import com.es.phoneshop.dao.Dao;
+import com.es.phoneshop.exception.ArgumentIsNullException;
+import com.es.phoneshop.exception.NotFoundException;
+import com.es.phoneshop.model.item.Item;
+
+import java.lang.reflect.ParameterizedType;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public abstract class GenericDao<T extends Item> implements Dao<T> {
```

Comment:
Please use Abstract prefix for abstract classes, this will make your code easier to understand